packages/server-wallet/src/__test__/create-and-close-channel/ledger-funding.test.ts

```diff
@@ -0,0 +1,590 @@
+        },
+      ],
+      token: ETH_ASSET_HOLDER_ADDRESS, // must be even length
+    };
+
+    const createChannelParams: CreateChannelParams = {
+      participants: [participantA, participantB],
+      allocations: [allocation],
+      appDefinition: ethers.constants.AddressZero,
+      appData: '0x00', // must be even length
+      fundingStrategy: 'Ledger',
+    };
+
+    //        A <> B
+    // PreFund0
+    const resultA0 = await a.createChannel(createChannelParams);
+
+    // TODO compute the channelId for a better test
+    const channelId = resultA0.channelResults[0].channelId;
+
+    expect(getChannelResultFor(channelId, resultA0.channelResults)).toMatchObject({
+      status: 'opening',
+      turnNum: 0,
+    });
+
+    expect(getSignedStateFor(channelId, resultA0.outbox)).toMatchObject(
+      {turnNum: 0} // The application's post fund
+    );
+
+    //    > PreFund0A
+    const resultB0 = await b.pushMessage(
+      getPayloadFor(participantB.participantId, resultA0.outbox)
+    );
+
+    expect(getChannelResultFor(channelId, resultB0.channelResults)).toMatchObject({
+      status: 'proposed',
+      turnNum: 0,
+    });
+
+    //       PreFund0B
+    //   LedgerUpdateB
+    const resultB1 = await b.joinChannel({channelId});
+
+    expect(getChannelResultFor(channelId, [resultB1.channelResult])).toMatchObject({
+      status: 'opening',
+      turnNum: 1,
+    });
+
+    expect(getSignedStateFor(ledgerChannelId, resultB1.outbox)).toMatchObject({turnNum: 5});
+
+    expect(getSignedStateFor(channelId, resultB1.outbox)).toMatchObject(
+      {turnNum: 1} // The application's pre fund
+    );
+
+    //        PreFund0B <
+    //    LedgerUpdateB <
+    // LedgerUpdateA
+    // PostFund2A
+    const resultA1 = await a.pushMessage(
+      getPayloadFor(participantA.participantId, resultB1.outbox)
+    );
+
+    expect(getChannelResultFor(channelId, resultA1.channelResults)).toMatchObject({
+      status: 'running',
+      turnNum: 2,
+    });
+
+    expect(getSignedStateFor(ledgerChannelId, resultA1.outbox)).toMatchObject({turnNum: 5});
+    expect(getSignedStateFor(channelId, resultA1.outbox)).toMatchObject(
+      {turnNum: 2} // The application's post fund
+    );
+
+    // > PostFund3A
+    // > LedgerUpdateA
+    //     PostFund3B
+    const resultB3 = await b.pushMessage(
+      getPayloadFor(participantB.participantId, resultA1.outbox)
+    );
+
+    expect(getChannelResultFor(channelId, resultB3.channelResults)).toMatchObject({
+      status: 'running',
+      turnNum: 3,
+    });
+
+    expect(getSignedStateFor(channelId, resultB3.outbox)).toMatchObject(
+      {turnNum: 3} // The application's post fund)
+    );
+
+    //    PostFund3B <
+    const resultA3 = await a.pushMessage(
+      getPayloadFor(participantA.participantId, resultB3.outbox)
+    );
+
+    expect(getChannelResultFor(channelId, resultA3.channelResults)).toMatchObject({
+      status: 'running',
+      turnNum: 3,
+    });
+  });
```

Comment:
I feel like this should assert on the final outcome of the channels.

Reply:
It asserts on the final outcome of the ledger channel — why would it need to assert on the outcome of the app channel?